core/commonMain/src/implementations/immutableMap/PersistentHashMapBuilder.kt

```diff
@@ -93,4 +95,43 @@ internal class PersistentHashMapBuilder<K, V>(private var map: PersistentHashMap
         node = TrieNode.EMPTY as TrieNode<K, V>
         size = 0
     }
+
+    private fun <K1, V1> containsEntry(entry: Map.Entry<K1, V1>): Boolean {
+        entry is Map.Entry<*, *> || return false
```

Comment:
We handle the same issue [here](https://github.com/Kotlin/kotlinx.collections.immutable/blob/79cbf64d64b928868599e00f5629574be793e9dc/core/commonMain/src/implementations/immutableMap/PersistentHashMapContentViews.kt#L14) and in 3 more locations of the project. Maybe we can extract the handling to a function? Or at least have the implementation everywhere.

Reply:
Hopefully, when the issue gets fixed or the legacy js backend is dropped, we can drop the workaround altogether.